.changes/next-release/enhancement-QueryCompatibilityModeHeader-5257.json

```
@@ -0,0 +1,5 @@
+{
+  "type": "enhancement",
+  "category": "Query Compatibility Mode Header",
+  "description": "Adds logic for custom header to support services migrating away from the AWS Query protocol."
```

super nit; We may consider avoiding wording like "Adding logic" or "custom header" in the release notes. Everything we write will probably fall under those categories so they're more filler words than informative. Especially for features like this that don't have a lot of external visibility to the end-user, it's good to keep the brief short and leave any additional details for the customer in the PR if needed.

```suggestion
  "description": "Added support for header enabling service migration off the AWS Query protocol."
```

or

```suggestion
  "description": "New header added to enable service migration off the AWS Query protocol."
```